### PR TITLE
Add a docker environment to run tests

### PR DESCRIPTION
**Pull request by wlach**

### CODE REVIEW

**Comment by wlach** [comment]
Not 100% sure about my approach here, but posting work-to-date nonetheless.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# Scala install based on hseeberger/scala-sbt
+
+ENV SCALA_VERSION 2.11.8
+ENV SBT_VERSION 0.13.13
+ENV HBASE_VERSION 1.2.3
+
+# Install Scala
+## Piping curl directly in tar
+RUN \
+  curl -fsL http://downloads.typesafe.com/scala/$SCALA_VERSION/scala-$SCALA_VERSION.tgz | tar xfz - -C /root/ && \
```
Comment (vitillo) [inline Dockerfile]: Scala should be installed automatically by `sbt` based on the configuration of `build.sbt`.

Comment (wlach) [inline Dockerfile]: Removed

**Comment by wlach** [comment]
Trying to figure out the most common workflow for running the tests by actually hacking on the code, will add documentation then.

**Comment by wlach** [comment]
@vitillo ok this is ready for another look. I added some documentation on running the tests. After working on an actual bug for telemetry-batch-view, I do think this is in fact useful for local development. 😄 

This blows away the caching we used to do for builds. I experimented a bit with caching the docker image, but it seemed rather complicated and the time taken by travis looked (by far) dominated by the time to test in any case.

File: runtests.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+set -e
+
+ARGS=$@
+
+# if we are not inside the docker container, run this command *inside* the
+# docker container
+if [ ! -f /.dockerenv ]; then
+    docker run -v $PWD:/telemetry-batch-view telemetry-batch-view ./runtests.sh
```
Comment (vitillo) [inline runtests.sh]: Every time this is re-run some JAR dependencies are refetched which is not optimal. Furthermore, it's not possible to run slow tests as the arguments aren't forwarded to `./runtests.sh`.

File: README.md [hunk 3]
```diff
@@ -16,7 +16,14 @@ See the [views](https://github.com/mozilla/telemetry-batch-view/tree/master/src/
 See the [docs](https://github.com/mozilla/telemetry-batch-view/tree/master/docs) folder for more information about the individual derived datasets.
 
 ### Development
-Before importing the project in IntelliJ IDEA, apply the following changes to `Preferences` -> `Languages & Frameworks` -> `Scala Compile Server`:
+There are two possible workflows for hacking on telemetry-batch-view: you can either create a docker container for building the package and running tests, or import the project into IntelliJ's IDEA.
+
+To run the docker tests, just use the provided `Dockerfile` to build a container, then use the `runtests.sh` script to run tests inside it:
```
Comment (wlach) [inline README.md]: Right, sorry, I remember having to do that too but forgot to right it down. I guess we could also fix https://bugzilla.mozilla.org/show_bug.cgi?id=1309214, which is probably the underlying cause of this requirement.

File: runtests.sh (hunk 2 above)
Comment (wlach) [inline runtests.sh]: Yeah, I'm more apt to be concerned about speed for the local case... should I use the same strategy we use in travis to cache the jar dependencies?

Comment (vitillo) [inline runtests.sh]: Not necessarily.

Comment (wlach) [inline runtests.sh]: Hmm, actually it *is* caching in the same way as the travis build already (under `/root/.sbt` and `/root/.ivy2`). But I do indeed see it redownloading a bunch of jar dependencies every time I launch runtests.sh:

```
Hammersmith:telemetry-batch-view wlach$ ./runtests.sh "test-only com.mozilla.telemetry.AddonsViewTest"
Picked up _JAVA_OPTIONS: -Xms4G -Xmx4G -Xss4M -XX:MaxMetaspaceSize=256M
starting master, logging to /hbase-1.2.3/bin/../logs/hbase--master-27f3a25037e0.out
OpenJDK 64-Bit Server VM warning: ignoring option PermSize=128m; support was removed in 8.0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=128m; support was removed in 8.0
Picked up _JAVA_OPTIONS: -Xms4G -Xmx4G -Xss4M -XX:MaxMetaspaceSize=256M
starting thrift, logging to /hbase-1.2.3/bin/../logs/hbase--thrift-27f3a25037e0.out
Picked up _JAVA_OPTIONS: -Xms4G -Xmx4G -Xss4M -XX:MaxMetaspaceSize=256M
Picked up _JAVA_OPTIONS: -Xms4G -Xmx4G -Xss4M -XX:MaxMetaspaceSize=256M
[info] Loading project definition from /telemetry-batch-view/project
[info] Updating {file:/telemetry-batch-view/project/}telemetry-batch-view-build...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] downloading https://repo1.maven.org/maven2/org/scoverage/sbt-scoverage_2.10_0.13/1.5.0/sbt-scoverage-1.5.0.jar ...
[info] 	[SUCCESSFUL ] org.scoverage#sbt-scoverage;1.5.0!sbt-scoverage.jar (95ms)
[info] downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.typesafe.sbteclipse/sbteclipse-plugin/scala_2.10/sbt_0.13/4.0.0/jars/sbteclipse-plugin.jar ...
[info] 	[SUCCESSFUL ] com.typesafe.sbteclipse#sbteclipse-plugin;4.0.0!sbteclipse-plugin.jar (1885ms)
[info] downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.eed3si9n/sbt-assembly/scala_2.10/sbt_0.13/0.14.0/jars/sbt-assembly.jar ...
[info] 	[SUCCESSFUL ] com.eed3si9n#sbt-assembly;0.14.0!sbt-assembly.jar (1469ms)
[info] downloading https://repo1.maven.org/maven2/org/scoverage/scalac-scoverage-plugin_2.10/1.3.0/scalac-scoverage-plugin_2.10-1.3.0.jar ...
[info] 	[SUCCESSFUL ] org.scoverage#scalac-scoverage-plugin_2.10;1.3.0!scalac-scoverage-plugin_2.10.jar (71ms)
[info] downloading https://repo1.maven.org/maven2/org/scalaz/scalaz-core_2.10/7.1.0/scalaz-core_2.10-7.1.0.jar ...
[info] 	[SUCCESSFUL ] org.scalaz#scalaz-core_2.10;7.1.0!scalaz-core_2.10.jar(bundle) (743ms)
[info] downloading https://repo1.maven.org/maven2/org/scalaz/scalaz-effect_2.10/7.1.0/scalaz-effect_2.10-7.1.0.jar ...
[info] 	[SUCCESSFUL ] org.scalaz#scalaz-effect_2.10;7.1.0!scalaz-effect_2.10.jar(bundle) (46ms)
[info] downloading https://repo1.maven.org/maven2/org/scalactic/scalactic_2.10/2.2.1/scalactic_2.10-2.2.1.jar ...
[info] 	[SUCCESSFUL ] org.scalactic#scalactic_2.10;2.2.1!scalactic_2.10.jar(bundle) (47ms)
[info] downloading https://repo1.maven.org/maven2/org/pantsbuild/jarjar/jarjar/1.5/jarjar-1.5.jar ...
[info] 	[SUCCESSFUL ] org.pantsbuild.jarjar#jarjar;1.5!jarjar.jar (28ms)
[info] Done updating.
```

I'll try to investigate what's going on.

Comment (wlach) [inline runtests.sh]: Think I figured it out -- the trick is to mount .ivy2 and .sbt into the docker directory from $HOME.

**Comment by vitillo** [comment]
@wlach The travis build is failing.

**Comment by wlach** [comment]
@vitillo this is ready for another look, sorry for the delay

File: README.md (hunk 3 above)
Comment (wlach) [inline README.md]: I think this is addressed now.

**Comment by vitillo** [comment]
@wlach: Please squash your changes into a single commit.

**Comment by wlach** [comment]
Yup, github has a feature which lets you do that easily. Will use it. :)

**Comment by vitillo** [comment]
We have disabled that feature to force contributors to adhere to our [guidelines](https://github.com/mozilla/telemetry-onboarding/blob/master/Contributing.md).

**Pull request merged**